From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jorge Vargas <yorch.veintemil+github@gmail.com>
Date: Sun, 18 Oct 2026 03:04:43 -0700
Subject: [PATCH] Patched for Raspberry Pi 5

Build the Talos v1.14.2 kernel package (pkgs v1.14.0-37-g6c312e4) from the
Raspberry Pi vendor kernel instead of kernel.org 6.18.54.

- Pkgfile: pin raspberrypi/linux stable_20260911, commit
  cff533aec2fa601846766b32ff57204e0a61bed7 (6.18.50), with its GitHub archive
  SHA-256/SHA-512.
- kernel/prepare: fetch that commit's GitHub archive (.tar.gz).
- kernel/build/config-arm64: the pkgs v1.14 config with the Raspberry Pi 5
  option delta carried by johnlaur/talos-builder (baseline 9e8a06cc) for
  Talos 1.13.2 re-applied, keeping SD/USB host/USB storage built in,
  disabling ARM64_BTI_KERNEL as 6.18.52+ does for clang >= 21, and keeping the
  Talos 1.14 options the 1.13 delta had turned off without a hardware reason.

Rebased from the "Patched for Raspberry Pi 5" patch of
johnlaur/talos-builder.
---
 Pkgfile                   |   8 +-
 kernel/build/config-arm64 | 433 ++++++++++++++++++++++++++------------
 kernel/prepare/pkg.yaml   |   6 +-
 3 files changed, 307 insertions(+), 140 deletions(-)

diff --git a/Pkgfile b/Pkgfile
index 42bc68f..9d15994 100644
--- a/Pkgfile
+++ b/Pkgfile
@@ -103,10 +103,10 @@ vars:
   kspp_sha256: 0dc61ebf01f0dd1d0d2204432235fb2470bcf2ec020172c30a9d1c14f103f032
   kspp_sha512: 2b40c41ed0347a339053aef61c838b5131dd583fb0d1ac8f3361040361ba868724ad16dfdc13023ff2b0f51f32c0aba605316d5d76f9fc6c1d50d90b4ceedead
 
-  # renovate: datasource=git-tags extractVersion=^v(?<version>.*)$ depName=git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
-  linux_version: 6.18.54
-  linux_sha256: 9df30b02dd8102bbd0be52556288ef6889ddbe7f1ddb96fbf847d0becf3eacac
-  linux_sha512: 37b4d8f6334345b769d2eccfb9a93e9d401b89ec582215d6a7035c94045ffc926fd332760d29e126285869c0b36c7d7c65a75c380d06ceb46bc15913a36009e3
+  # rpi-6.18.y branch, commit cff533aec2fa (2026-09-10), kernel 6.18.50, tag stable_20260911
+  linux_version: cff533aec2fa601846766b32ff57204e0a61bed7
+  linux_sha256: 6c58de91ade97d7cd9176e408f05c9d134cda64dba95a87b7af976bf26c0c0bd
+  linux_sha512: 6f491c45247df1fef7481811c16975fd3ff75d89497e3b748d5f52fa9e28d9cb5d6f10b3ffa30265289d6599ba2d252cd486ca19e1ea94d24291684f6c14aef6
 
   # renovate: datasource=git-tags extractVersion=^libaio-(?<version>.*)$ depName=https://pagure.io/libaio.git
   libaio_version: 0.3.113
diff --git a/kernel/build/config-arm64 b/kernel/build/config-arm64
index b675aad..a60b6d1 100644
--- a/kernel/build/config-arm64
+++ b/kernel/build/config-arm64
@@ -294,6 +294,7 @@ CONFIG_TRACEPOINTS=y
 CONFIG_CRASH_RESERVE=y
 CONFIG_VMCORE_INFO=y
 CONFIG_KEXEC_CORE=y
+CONFIG_HAVE_IMA_KEXEC=y
 # CONFIG_KEXEC is not set
 CONFIG_KEXEC_FILE=y
 CONFIG_KEXEC_SIG=y
@@ -457,8 +458,8 @@ CONFIG_ROCKCHIP_ERRATUM_3588001=y
 CONFIG_SOCIONEXT_SYNQUACER_PREITS=y
 # end of ARM errata workarounds via the alternatives framework
 
-CONFIG_ARM64_4K_PAGES=y
-# CONFIG_ARM64_16K_PAGES is not set
+# CONFIG_ARM64_4K_PAGES is not set
+CONFIG_ARM64_16K_PAGES=y
 # CONFIG_ARM64_64K_PAGES is not set
 # CONFIG_ARM64_VA_BITS_39 is not set
 CONFIG_ARM64_VA_BITS_48=y
@@ -466,6 +467,7 @@ CONFIG_ARM64_VA_BITS_48=y
 CONFIG_ARM64_VA_BITS=48
 CONFIG_ARM64_PA_BITS_48=y
 CONFIG_ARM64_PA_BITS=48
+# CONFIG_CPU_BIG_ENDIAN is not set
 CONFIG_CPU_LITTLE_ENDIAN=y
 CONFIG_NR_CPUS=4096
 CONFIG_HOTPLUG_CPU=y
@@ -498,6 +500,7 @@ CONFIG_XEN=y
 CONFIG_ARCH_FORCE_MAX_ORDER=10
 CONFIG_UNMAP_KERNEL_AT_EL0=y
 CONFIG_MITIGATE_SPECTRE_BRANCH_HISTORY=y
+CONFIG_RODATA_FULL_DEFAULT_ENABLED=y
 CONFIG_ARM64_SW_TTBR0_PAN=y
 CONFIG_ARM64_TAGGED_ADDR_ABI=y
 CONFIG_COMPAT=y
@@ -547,6 +550,7 @@ CONFIG_ARM64_TLB_RANGE=y
 #
 CONFIG_AS_HAS_ARMV8_5=y
 CONFIG_ARM64_BTI=y
+# CONFIG_ARM64_BTI_KERNEL is not set
 CONFIG_CC_HAS_BRANCH_PROT_PAC_RET_BTI=y
 CONFIG_ARM64_E0PD=y
 CONFIG_ARM64_AS_HAS_MTE=y
@@ -862,6 +866,7 @@ CONFIG_LTO_CLANG_THIN=y
 CONFIG_ARCH_SUPPORTS_CFI=y
 # CONFIG_CFI is not set
 CONFIG_HAVE_CFI_ICALL_NORMALIZE_INTEGERS=y
+CONFIG_ARCH_SUPPORTS_CFI_CLANG=y
 CONFIG_HAVE_CONTEXT_TRACKING_USER=y
 CONFIG_HAVE_VIRT_CPU_ACCOUNTING_GEN=y
 CONFIG_HAVE_IRQ_TIME_ACCOUNTING=y
@@ -932,6 +937,7 @@ CONFIG_HAVE_GCC_PLUGINS=y
 CONFIG_FUNCTION_ALIGNMENT_4B=y
 CONFIG_FUNCTION_ALIGNMENT_8B=y
 CONFIG_FUNCTION_ALIGNMENT=8
+CONFIG_CC_HAS_MIN_FUNCTION_ALIGNMENT=y
 CONFIG_CC_HAS_SANE_FUNCTION_ALIGNMENT=y
 CONFIG_HAVE_ARCH_GET_SECUREBOOT=y
 # end of General architecture-dependent options
@@ -946,12 +952,13 @@ CONFIG_MODULE_UNLOAD=y
 # CONFIG_MODULE_UNLOAD_TAINT_TRACKING is not set
 CONFIG_MODVERSIONS=y
 CONFIG_GENKSYMS=y
+# CONFIG_GENDWARFKSYMS is not set
 CONFIG_ASM_MODVERSIONS=y
 # CONFIG_EXTENDED_MODVERSIONS is not set
 CONFIG_BASIC_MODVERSIONS=y
 CONFIG_MODULE_SRCVERSION_ALL=y
 CONFIG_MODULE_SIG=y
-# CONFIG_MODULE_SIG_FORCE is not set
+CONFIG_MODULE_SIG_FORCE=y
 CONFIG_MODULE_SIG_ALL=y
 # CONFIG_MODULE_SIG_SHA1 is not set
 # CONFIG_MODULE_SIG_SHA256 is not set
@@ -1299,6 +1306,7 @@ CONFIG_XFRM_ESP=y
 CONFIG_XFRM_IPCOMP=y
 # CONFIG_NET_KEY is not set
 # CONFIG_XFRM_IPTFS is not set
+# CONFIG_SMC is not set
 # CONFIG_DIBS is not set
 CONFIG_XDP_SOCKETS=y
 # CONFIG_XDP_SOCKETS_DIAG is not set
@@ -1402,7 +1410,7 @@ CONFIG_NETLABEL=y
 # CONFIG_MPTCP is not set
 CONFIG_NETWORK_SECMARK=y
 CONFIG_NET_PTP_CLASSIFY=y
-# CONFIG_NETWORK_PHY_TIMESTAMPING is not set
+CONFIG_NETWORK_PHY_TIMESTAMPING=y
 CONFIG_NETFILTER=y
 CONFIG_NETFILTER_ADVANCED=y
 CONFIG_BRIDGE_NETFILTER=y
@@ -1714,8 +1722,12 @@ CONFIG_NF_TABLES_BRIDGE=y
 # CONFIG_BRIDGE_NF_EBTABLES is not set
 CONFIG_IP_SCTP=y
 # CONFIG_SCTP_DBG_OBJCNT is not set
+CONFIG_SCTP_DEFAULT_COOKIE_HMAC_MD5=y
+# CONFIG_SCTP_DEFAULT_COOKIE_HMAC_SHA1 is not set
 CONFIG_SCTP_DEFAULT_COOKIE_HMAC_SHA256=y
 # CONFIG_SCTP_DEFAULT_COOKIE_HMAC_NONE is not set
+CONFIG_SCTP_COOKIE_HMAC_MD5=y
+# CONFIG_SCTP_COOKIE_HMAC_SHA1 is not set
 CONFIG_INET_SCTP_DIAG=y
 # CONFIG_RDS is not set
 # CONFIG_TIPC is not set
@@ -2047,7 +2059,7 @@ CONFIG_PCIE_DW=y
 CONFIG_PCIE_DW_HOST=y
 # CONFIG_PCIE_AL is not set
 # CONFIG_PCIE_AMD_MDB is not set
-CONFIG_PCI_MESON=y
+# CONFIG_PCI_MESON is not set
 # CONFIG_PCI_IMX6_HOST is not set
 # CONFIG_PCI_IMX6_EP is not set
 CONFIG_PCI_LAYERSCAPE=y
@@ -2105,7 +2117,7 @@ CONFIG_PCI_EPF_NTB=y
 
 CONFIG_HAVE_PWRCTRL=y
 CONFIG_PCI_PWRCTRL=y
-CONFIG_PCI_PWRCTRL_SLOT=y
+# CONFIG_PCI_PWRCTRL_SLOT is not set
 # CONFIG_CXL_BUS is not set
 # CONFIG_PCCARD is not set
 # CONFIG_RAPIDIO is not set
@@ -2241,6 +2253,7 @@ CONFIG_TURRIS_MOX_RWTM=y
 CONFIG_TURRIS_MOX_RWTM_KEYCTL=y
 # CONFIG_ARM_FFA_TRANSPORT is not set
 # CONFIG_GOOGLE_FIRMWARE is not set
+CONFIG_FIRMWARE_RP1=y
 
 #
 # EFI (Extensible Firmware Interface) Support
@@ -2424,6 +2437,7 @@ CONFIG_OF_RESERVED_MEM=y
 CONFIG_OF_RESOLVE=y
 CONFIG_OF_OVERLAY=y
 CONFIG_OF_NUMA=y
+# CONFIG_OF_CONFIGFS is not set
 # CONFIG_PARPORT is not set
 CONFIG_PNP=y
 CONFIG_PNP_DEBUG_MESSAGES=y
@@ -2456,7 +2470,7 @@ CONFIG_BLK_DEV_UBLK=m
 #
 CONFIG_NVME_AUTH=m
 CONFIG_NVME_CORE=y
-CONFIG_BLK_DEV_NVME=m
+CONFIG_BLK_DEV_NVME=y
 CONFIG_NVME_MULTIPATH=y
 # CONFIG_NVME_VERBOSE_ERRORS is not set
 CONFIG_NVME_HWMON=y
@@ -2477,6 +2491,7 @@ CONFIG_NVME_TARGET_TCP=m
 # CONFIG_NVME_TARGET_TCP_TLS is not set
 CONFIG_NVME_TARGET_AUTH=y
 # CONFIG_NVME_TARGET_PCI_EPF is not set
+CONFIG_BCM2835_SMI=m
 # end of NVME Support
 
 #
@@ -2491,7 +2506,7 @@ CONFIG_NVME_TARGET_AUTH=y
 # CONFIG_ICS932S401 is not set
 CONFIG_ENCLOSURE_SERVICES=y
 # CONFIG_HI6421V600_IRQ is not set
-CONFIG_HP_ILO=m
+CONFIG_HP_ILO=y
 # CONFIG_QCOM_COINCELL is not set
 # CONFIG_QCOM_FASTRPC is not set
 # CONFIG_APDS9802ALS is not set
@@ -2515,6 +2530,8 @@ CONFIG_SRAM=y
 # CONFIG_MARVELL_CN10K_DPI is not set
 # CONFIG_MCHP_LAN966X_PCI is not set
 # CONFIG_C2PORT is not set
+CONFIG_RP1_PIO=y
+CONFIG_WS2812_PIO_RP1=y
 
 #
 # EEPROM support
@@ -2543,7 +2560,7 @@ CONFIG_EEPROM_93CX6=y
 # CONFIG_GP_PCI1XXXX is not set
 # CONFIG_KEBA_CP500 is not set
 # CONFIG_AMD_SBRMI_I2C is not set
-CONFIG_MISC_RP1=y
+# CONFIG_MISC_RP1 is not set
 # end of Misc devices
 
 #
@@ -2593,7 +2610,7 @@ CONFIG_ISCSI_BOOT_SYSFS=y
 # CONFIG_SCSI_BNX2X_FCOE is not set
 # CONFIG_BE2ISCSI is not set
 # CONFIG_BLK_DEV_3W_XXXX_RAID is not set
-CONFIG_SCSI_HPSA=m
+CONFIG_SCSI_HPSA=y
 # CONFIG_SCSI_3W_9XXX is not set
 # CONFIG_SCSI_3W_SAS is not set
 # CONFIG_SCSI_ACARD is not set
@@ -2611,13 +2628,13 @@ CONFIG_SCSI_HISI_SAS_PCI=m
 # CONFIG_SCSI_ESAS2R is not set
 # CONFIG_MEGARAID_NEWGEN is not set
 # CONFIG_MEGARAID_LEGACY is not set
-CONFIG_MEGARAID_SAS=m
-CONFIG_SCSI_MPT3SAS=m
+CONFIG_MEGARAID_SAS=y
+CONFIG_SCSI_MPT3SAS=y
 CONFIG_SCSI_MPT2SAS_MAX_SGE=128
 CONFIG_SCSI_MPT3SAS_MAX_SGE=128
 # CONFIG_SCSI_MPT2SAS is not set
 CONFIG_SCSI_MPI3MR=m
-CONFIG_SCSI_SMARTPQI=m
+CONFIG_SCSI_SMARTPQI=y
 # CONFIG_SCSI_HPTIOP is not set
 # CONFIG_SCSI_BUSLOGIC is not set
 # CONFIG_SCSI_MYRB is not set
@@ -2673,7 +2690,7 @@ CONFIG_SATA_PMP=y
 #
 # Controllers with non-SFF native interface
 #
-CONFIG_SATA_AHCI=m
+CONFIG_SATA_AHCI=y
 CONFIG_SATA_MOBILE_LPM_POLICY=0
 CONFIG_SATA_AHCI_PLATFORM=y
 # CONFIG_AHCI_BRCM is not set
@@ -2720,7 +2737,7 @@ CONFIG_SATA_RCAR=y
 # PATA SFF controllers with BMDMA
 #
 # CONFIG_PATA_ALI is not set
-CONFIG_PATA_AMD=m
+CONFIG_PATA_AMD=y
 # CONFIG_PATA_ARTOP is not set
 # CONFIG_PATA_ATIIXP is not set
 # CONFIG_PATA_ATP867X is not set
@@ -2735,17 +2752,17 @@ CONFIG_PATA_AMD=m
 # CONFIG_PATA_IT8213 is not set
 # CONFIG_PATA_IT821X is not set
 # CONFIG_PATA_JMICRON is not set
-CONFIG_PATA_MARVELL=m
+CONFIG_PATA_MARVELL=y
 # CONFIG_PATA_NETCELL is not set
 # CONFIG_PATA_NINJA32 is not set
 # CONFIG_PATA_NS87415 is not set
-CONFIG_PATA_OLDPIIX=m
+CONFIG_PATA_OLDPIIX=y
 # CONFIG_PATA_OPTIDMA is not set
 # CONFIG_PATA_PDC2027X is not set
 # CONFIG_PATA_PDC_OLD is not set
 # CONFIG_PATA_RADISYS is not set
 # CONFIG_PATA_RDC is not set
-CONFIG_PATA_SCH=m
+CONFIG_PATA_SCH=y
 # CONFIG_PATA_SERVERWORKS is not set
 # CONFIG_PATA_SIL680 is not set
 # CONFIG_PATA_SIS is not set
@@ -2782,7 +2799,7 @@ CONFIG_MD_RAID0=y
 CONFIG_MD_RAID1=y
 CONFIG_MD_RAID10=y
 CONFIG_MD_RAID456=m
-CONFIG_BCACHE=m
+CONFIG_BCACHE=y
 # CONFIG_BCACHE_DEBUG is not set
 # CONFIG_BCACHE_ASYNC_REGISTRATION is not set
 CONFIG_BLK_DEV_DM_BUILTIN=y
@@ -2908,7 +2925,7 @@ CONFIG_NET_VRF=m
 # end of Distributed Switch Architecture drivers
 
 CONFIG_ETHERNET=y
-CONFIG_MDIO=m
+CONFIG_MDIO=y
 CONFIG_NET_VENDOR_3COM=y
 # CONFIG_VORTEX is not set
 # CONFIG_TYPHOON is not set
@@ -2936,7 +2953,7 @@ CONFIG_AMD_XGBE=m
 CONFIG_NET_XGENE=y
 # CONFIG_NET_XGENE_V2 is not set
 CONFIG_NET_VENDOR_AQUANTIA=y
-CONFIG_AQTION=m
+CONFIG_AQTION=y
 CONFIG_NET_VENDOR_ARC=y
 # CONFIG_EMAC_ROCKCHIP is not set
 CONFIG_NET_VENDOR_ASIX=y
@@ -2950,16 +2967,16 @@ CONFIG_ALX=m
 CONFIG_NET_VENDOR_BROADCOM=y
 # CONFIG_B44 is not set
 CONFIG_BCMGENET=y
-CONFIG_BNX2=m
+CONFIG_BNX2=y
 # CONFIG_CNIC is not set
-CONFIG_TIGON3=m
+CONFIG_TIGON3=y
 CONFIG_TIGON3_HWMON=y
-CONFIG_BNX2X=m
+CONFIG_BNX2X=y
 CONFIG_BNX2X_SRIOV=y
 CONFIG_BGMAC=y
 CONFIG_BGMAC_PLATFORM=y
 # CONFIG_SYSTEMPORT is not set
-CONFIG_BNXT=m
+CONFIG_BNXT=y
 CONFIG_BNXT_SRIOV=y
 CONFIG_BNXT_FLOWER_OFFLOAD=y
 # CONFIG_BNXT_DCB is not set
@@ -2975,7 +2992,7 @@ CONFIG_NET_VENDOR_CAVIUM=y
 # CONFIG_THUNDER_NIC_VF is not set
 # CONFIG_THUNDER_NIC_BGX is not set
 # CONFIG_THUNDER_NIC_RGX is not set
-CONFIG_CAVIUM_PTP=m
+CONFIG_CAVIUM_PTP=y
 # CONFIG_LIQUIDIO is not set
 # CONFIG_LIQUIDIO_VF is not set
 CONFIG_NET_VENDOR_CHELSIO=y
@@ -2988,7 +3005,7 @@ CONFIG_CHELSIO_T4VF=m
 CONFIG_CHELSIO_INLINE_CRYPTO=y
 CONFIG_CHELSIO_IPSEC_INLINE=m
 CONFIG_NET_VENDOR_CISCO=y
-CONFIG_ENIC=m
+CONFIG_ENIC=y
 CONFIG_NET_VENDOR_CORTINA=y
 # CONFIG_GEMINI_ETHERNET is not set
 CONFIG_NET_VENDOR_DAVICOM=y
@@ -3046,32 +3063,32 @@ CONFIG_NET_VENDOR_HUAWEI=y
 # CONFIG_HINIC3 is not set
 CONFIG_NET_VENDOR_I825XX=y
 CONFIG_NET_VENDOR_INTEL=y
-CONFIG_LIBETH=m
+CONFIG_LIBETH=y
 CONFIG_LIBETH_XDP=m
-CONFIG_LIBIE=m
+CONFIG_LIBIE=y
 CONFIG_LIBIE_ADMINQ=m
 CONFIG_LIBIE_FWLOG=m
-CONFIG_E100=m
-CONFIG_E1000=m
-CONFIG_E1000E=m
-CONFIG_IGB=m
+CONFIG_E100=y
+CONFIG_E1000=y
+CONFIG_E1000E=y
+CONFIG_IGB=y
 CONFIG_IGB_HWMON=y
-CONFIG_IGBVF=m
-CONFIG_IXGBE=m
+CONFIG_IGBVF=y
+CONFIG_IXGBE=y
 CONFIG_IXGBE_HWMON=y
 # CONFIG_IXGBE_DCB is not set
 CONFIG_IXGBE_IPSEC=y
-CONFIG_IXGBEVF=m
+CONFIG_IXGBEVF=y
 CONFIG_IXGBEVF_IPSEC=y
-CONFIG_I40E=m
+CONFIG_I40E=y
 # CONFIG_I40E_DCB is not set
-CONFIG_IAVF=m
-CONFIG_I40EVF=m
-CONFIG_ICE=m
+CONFIG_IAVF=y
+CONFIG_I40EVF=y
+CONFIG_ICE=y
 CONFIG_ICE_HWMON=y
 CONFIG_ICE_SWITCHDEV=y
 # CONFIG_FM10K is not set
-CONFIG_IGC=m
+CONFIG_IGC=y
 CONFIG_IDPF=m
 # CONFIG_IDPF_SINGLEQ is not set
 # CONFIG_JME is not set
@@ -3086,7 +3103,7 @@ CONFIG_MVPP2=y
 # CONFIG_MVPP2_PTP is not set
 # CONFIG_PXA168_ETH is not set
 # CONFIG_SKGE is not set
-CONFIG_SKY2=m
+CONFIG_SKY2=y
 # CONFIG_SKY2_DEBUG is not set
 # CONFIG_OCTEONTX2_AF is not set
 # CONFIG_OCTEONTX2_PF is not set
@@ -3171,15 +3188,15 @@ CONFIG_NET_VENDOR_PENSANDO=y
 # CONFIG_IONIC is not set
 CONFIG_NET_VENDOR_QLOGIC=y
 # CONFIG_QLA3XXX is not set
-CONFIG_QLCNIC=m
+CONFIG_QLCNIC=y
 CONFIG_QLCNIC_SRIOV=y
 CONFIG_QLCNIC_DCB=y
 CONFIG_QLCNIC_HWMON=y
 # CONFIG_NETXEN_NIC is not set
-CONFIG_QED=m
+CONFIG_QED=y
 CONFIG_QED_LL2=y
 CONFIG_QED_SRIOV=y
-CONFIG_QEDE=m
+CONFIG_QEDE=y
 CONFIG_QED_FCOE=y
 CONFIG_NET_VENDOR_BROCADE=y
 # CONFIG_BNA is not set
@@ -3192,12 +3209,12 @@ CONFIG_NET_VENDOR_RDC=y
 # CONFIG_R6040 is not set
 CONFIG_NET_VENDOR_REALTEK=y
 # CONFIG_8139CP is not set
-CONFIG_8139TOO=m
+CONFIG_8139TOO=y
 CONFIG_8139TOO_PIO=y
 # CONFIG_8139TOO_TUNE_TWISTER is not set
 # CONFIG_8139TOO_8129 is not set
 # CONFIG_8139_OLD_RX_RESET is not set
-CONFIG_R8169=m
+CONFIG_R8169=y
 # CONFIG_RTASE is not set
 CONFIG_NET_VENDOR_RENESAS=y
 # CONFIG_SH_ETH is not set
@@ -3234,24 +3251,24 @@ CONFIG_SMSC911X=y
 CONFIG_NET_VENDOR_SOCIONEXT=y
 # CONFIG_SNI_NETSEC is not set
 CONFIG_NET_VENDOR_STMICRO=y
-CONFIG_STMMAC_ETH=m
+CONFIG_STMMAC_ETH=y
 # CONFIG_STMMAC_SELFTESTS is not set
-CONFIG_STMMAC_PLATFORM=m
-CONFIG_DWMAC_DWC_QOS_ETH=m
-CONFIG_DWMAC_GENERIC=m
-CONFIG_DWMAC_IPQ806X=m
+CONFIG_STMMAC_PLATFORM=y
+CONFIG_DWMAC_DWC_QOS_ETH=y
+CONFIG_DWMAC_GENERIC=y
+CONFIG_DWMAC_IPQ806X=y
 # CONFIG_DWMAC_MEDIATEK is not set
-CONFIG_DWMAC_MESON=m
-CONFIG_DWMAC_QCOM_ETHQOS=m
+CONFIG_DWMAC_MESON=y
+CONFIG_DWMAC_QCOM_ETHQOS=y
 CONFIG_DWMAC_RENESAS_GBETH=m
-CONFIG_DWMAC_ROCKCHIP=m
-CONFIG_DWMAC_SUNXI=m
-CONFIG_DWMAC_SUN8I=m
+CONFIG_DWMAC_ROCKCHIP=y
+CONFIG_DWMAC_SUNXI=y
+CONFIG_DWMAC_SUN8I=y
 CONFIG_DWMAC_SUN55I=m
-CONFIG_DWMAC_IMX8=m
+CONFIG_DWMAC_IMX8=y
 # CONFIG_DWMAC_INTEL_PLAT is not set
-CONFIG_DWMAC_TEGRA=m
-CONFIG_STMMAC_PCI=m
+CONFIG_DWMAC_TEGRA=y
+CONFIG_STMMAC_PCI=y
 CONFIG_NET_VENDOR_SUN=y
 # CONFIG_HAPPYMEAL is not set
 # CONFIG_SUNGEM is not set
@@ -3355,12 +3372,13 @@ CONFIG_TERANETICS_PHY=y
 CONFIG_DP83822_PHY=y
 CONFIG_DP83TC811_PHY=y
 CONFIG_DP83848_PHY=y
-CONFIG_DP83867_PHY=m
+CONFIG_DP83867_PHY=y
 CONFIG_DP83869_PHY=y
 # CONFIG_DP83TD510_PHY is not set
 # CONFIG_DP83TG720_PHY is not set
 CONFIG_VITESSE_PHY=y
 # CONFIG_XILINX_GMII2RGMII is not set
+# CONFIG_MICREL_KS8995MA is not set
 # CONFIG_PSE_CONTROLLER is not set
 CONFIG_FWNODE_MDIO=y
 CONFIG_OF_MDIO=y
@@ -3394,7 +3412,7 @@ CONFIG_MDIO_BUS_MUX_MMIOREG=y
 #
 # PCS device drivers
 #
-CONFIG_PCS_XPCS=m
+CONFIG_PCS_XPCS=y
 CONFIG_PCS_RZN1_MIIC=m
 # end of PCS device drivers
 
@@ -3417,7 +3435,7 @@ CONFIG_USB_NET_DRIVERS=y
 # CONFIG_USB_KAWETH is not set
 CONFIG_USB_PEGASUS=y
 CONFIG_USB_RTL8150=y
-CONFIG_USB_RTL8152=m
+CONFIG_USB_RTL8152=y
 CONFIG_USB_LAN78XX=y
 CONFIG_USB_USBNET=m
 CONFIG_USB_NET_AX8817X=m
@@ -3450,7 +3468,7 @@ CONFIG_USB_NET_CX82310_ETH=m
 CONFIG_USB_NET_KALMIA=m
 CONFIG_USB_NET_QMI_WWAN=m
 CONFIG_USB_NET_INT51X1=m
-CONFIG_USB_IPHETH=m
+# CONFIG_USB_IPHETH is not set
 CONFIG_USB_SIERRA_NET=m
 CONFIG_USB_VL600=m
 CONFIG_USB_NET_CH9200=m
@@ -3466,7 +3484,7 @@ CONFIG_USB_RTL8153_ECM=m
 # end of Wireless WAN
 
 CONFIG_XEN_NETDEV_FRONTEND=m
-CONFIG_VMXNET3=m
+CONFIG_VMXNET3=y
 # CONFIG_FUJITSU_ES is not set
 CONFIG_USB4_NET=m
 CONFIG_HYPERV_NET=m
@@ -3491,7 +3509,7 @@ CONFIG_INPUT_MOUSEDEV=y
 # CONFIG_INPUT_MOUSEDEV_PSAUX is not set
 CONFIG_INPUT_MOUSEDEV_SCREEN_X=1024
 CONFIG_INPUT_MOUSEDEV_SCREEN_Y=768
-CONFIG_INPUT_JOYDEV=m
+# CONFIG_INPUT_JOYDEV is not set
 CONFIG_INPUT_EVDEV=y
 
 #
@@ -3508,6 +3526,7 @@ CONFIG_KEYBOARD_ATKBD=y
 # CONFIG_KEYBOARD_LKKBD is not set
 # CONFIG_KEYBOARD_GPIO is not set
 # CONFIG_KEYBOARD_GPIO_POLLED is not set
+# CONFIG_KEYBOARD_TCA6416 is not set
 # CONFIG_KEYBOARD_TCA8418 is not set
 # CONFIG_KEYBOARD_MATRIX is not set
 # CONFIG_KEYBOARD_LM8323 is not set
@@ -3733,6 +3752,8 @@ CONFIG_HYPERV_KEYBOARD=m
 # CONFIG_SERIO_GPIO_PS2 is not set
 # CONFIG_USERIO is not set
 # CONFIG_GAMEPORT is not set
+# CONFIG_BRCM_CHAR_DRIVERS is not set
+CONFIG_BCM2835_SMI_DEV=m
 # end of Hardware I/O ports
 # end of Input device support
 
@@ -3831,6 +3852,7 @@ CONFIG_SERIAL_XILINX_PS_UART_CONSOLE=y
 CONFIG_SERIAL_MVEBU_UART=y
 CONFIG_SERIAL_MVEBU_CONSOLE=y
 # CONFIG_SERIAL_OWL is not set
+# CONFIG_SERIAL_RPI_FW is not set
 # end of Serial drivers
 
 CONFIG_SERIAL_MCTRL_GPIO=y
@@ -3906,6 +3928,7 @@ CONFIG_TCG_CRB=y
 # CONFIG_TCG_TIS_ST33ZP24_SPI is not set
 # CONFIG_XILLYBUS is not set
 # CONFIG_XILLYUSB is not set
+CONFIG_RASPBERRYPI_GPIOMEM=y
 # end of Character devices
 
 #
@@ -3915,7 +3938,7 @@ CONFIG_I2C=y
 CONFIG_ACPI_I2C_OPREGION=y
 CONFIG_I2C_BOARDINFO=y
 CONFIG_I2C_CHARDEV=y
-CONFIG_I2C_MUX=m
+CONFIG_I2C_MUX=y
 
 #
 # Multiplexer I2C Chip support
@@ -3934,7 +3957,7 @@ CONFIG_I2C_MUX=m
 
 CONFIG_I2C_HELPER_AUTO=y
 CONFIG_I2C_SMBUS=y
-CONFIG_I2C_ALGOBIT=m
+CONFIG_I2C_ALGOBIT=y
 
 #
 # I2C Hardware Bus support
@@ -3944,6 +3967,7 @@ CONFIG_I2C_ALGOBIT=m
 # PC SMBus host controller drivers
 #
 CONFIG_I2C_CCGX_UCSI=y
+CONFIG_I2C_BCM2708=y
 # CONFIG_I2C_ALI1535 is not set
 # CONFIG_I2C_ALI1563 is not set
 # CONFIG_I2C_ALI15X3 is not set
@@ -3951,7 +3975,7 @@ CONFIG_I2C_CCGX_UCSI=y
 # CONFIG_I2C_AMD8111 is not set
 # CONFIG_I2C_AMD_MP2 is not set
 # CONFIG_I2C_HIX5HD2 is not set
-CONFIG_I2C_I801=m
+CONFIG_I2C_I801=y
 # CONFIG_I2C_ISCH is not set
 # CONFIG_I2C_PIIX4 is not set
 # CONFIG_I2C_NFORCE2 is not set
@@ -4068,12 +4092,13 @@ CONFIG_SPI_BCM_QSPI=y
 # CONFIG_SPI_HISI_KUNPENG is not set
 # CONFIG_SPI_HISI_SFC_V3XX is not set
 # CONFIG_SPI_NXP_FLEXSPI is not set
-# CONFIG_SPI_GPIO is not set
+CONFIG_SPI_GPIO=y
 # CONFIG_SPI_IMX is not set
 # CONFIG_SPI_FSL_SPI is not set
 # CONFIG_SPI_FSL_DSPI is not set
 CONFIG_SPI_MESON_SPICC=y
 CONFIG_SPI_MESON_SPIFC=y
+# CONFIG_SPI_MICROCHIP_CORE is not set
 # CONFIG_SPI_MICROCHIP_CORE_QSPI is not set
 # CONFIG_SPI_MT65XX is not set
 # CONFIG_SPI_MTK_NOR is not set
@@ -4084,6 +4109,7 @@ CONFIG_SPI_ORION=y
 CONFIG_SPI_PL022=y
 CONFIG_SPI_ROCKCHIP=y
 # CONFIG_SPI_ROCKCHIP_SFC is not set
+CONFIG_SPI_RP2040_GPIO_BRIDGE=y
 # CONFIG_SPI_RSPI is not set
 # CONFIG_SPI_RZV2H_RSPI is not set
 # CONFIG_SPI_RZV2M_CSI is not set
@@ -4182,7 +4208,7 @@ CONFIG_PINCTRL_AXP209=y
 CONFIG_PINCTRL_BM1880=y
 # CONFIG_PINCTRL_CY8C95X0 is not set
 CONFIG_PINCTRL_MAX77620=y
-# CONFIG_PINCTRL_MCP23S08 is not set
+CONFIG_PINCTRL_MCP23S08=y
 # CONFIG_PINCTRL_MICROCHIP_SGPIO is not set
 # CONFIG_PINCTRL_OCELOT is not set
 CONFIG_PINCTRL_RK805=y
@@ -4378,6 +4404,7 @@ CONFIG_GPIO_GENERIC=y
 # CONFIG_GPIO_AMDPT is not set
 CONFIG_GPIO_RASPBERRYPI_EXP=y
 CONFIG_GPIO_BCM_XGS_IPROC=y
+CONFIG_GPIO_BCM_VIRT=y
 CONFIG_GPIO_BRCMSTB=y
 # CONFIG_GPIO_CADENCE is not set
 # CONFIG_GPIO_DAVINCI is not set
@@ -4395,6 +4422,7 @@ CONFIG_GPIO_DWAPB=y
 CONFIG_GPIO_MVEBU=y
 CONFIG_GPIO_MXC=y
 CONFIG_GPIO_PL061=y
+# CONFIG_GPIO_PWM is not set
 # CONFIG_GPIO_POLARFIRE_SOC is not set
 CONFIG_GPIO_RCAR=y
 CONFIG_GPIO_ROCKCHIP=y
@@ -4429,6 +4457,7 @@ CONFIG_GPIO_PCA953X_IRQ=y
 # CONFIG_GPIO_PCA9570 is not set
 # CONFIG_GPIO_PCF857X is not set
 # CONFIG_GPIO_TPIC2810 is not set
+# CONFIG_GPIO_FSM is not set
 # end of I2C GPIO expanders
 
 #
@@ -4503,6 +4532,7 @@ CONFIG_POWER_RESET_SYSCON=y
 CONFIG_POWER_SUPPLY=y
 # CONFIG_POWER_SUPPLY_DEBUG is not set
 CONFIG_POWER_SUPPLY_HWMON=y
+# CONFIG_RPI_POE_POWER is not set
 # CONFIG_GENERIC_ADC_BATTERY is not set
 # CONFIG_IP5XXX_POWER is not set
 # CONFIG_TEST_POWER is not set
@@ -4604,7 +4634,7 @@ CONFIG_SENSORS_I5K_AMB=m
 # CONFIG_SENSORS_GL520SM is not set
 # CONFIG_SENSORS_G760A is not set
 # CONFIG_SENSORS_G762 is not set
-CONFIG_SENSORS_GPIO_FAN=y
+# CONFIG_SENSORS_GPIO_FAN is not set
 # CONFIG_SENSORS_HIH6130 is not set
 # CONFIG_SENSORS_HS3001 is not set
 # CONFIG_SENSORS_HTU31 is not set
@@ -4788,7 +4818,7 @@ CONFIG_AMLOGIC_THERMAL=y
 # Broadcom thermal drivers
 #
 CONFIG_BCM2711_THERMAL=y
-# CONFIG_BCM2835_THERMAL is not set
+CONFIG_BCM2835_THERMAL=y
 # CONFIG_BRCMSTB_THERMAL is not set
 CONFIG_BCM_NS_THERMAL=y
 CONFIG_BCM_SR_THERMAL=y
@@ -4967,6 +4997,7 @@ CONFIG_MFD_MAX77620=y
 # CONFIG_MFD_QCOM_RPM is not set
 CONFIG_MFD_SPMI_PMIC=y
 # CONFIG_MFD_SY7636A is not set
+CONFIG_MFD_RASPBERRYPI_POE_HAT=y
 # CONFIG_MFD_RDC321X is not set
 # CONFIG_MFD_RT4831 is not set
 # CONFIG_MFD_RT5033 is not set
@@ -5031,6 +5062,7 @@ CONFIG_MFD_SYSCON=y
 # CONFIG_MFD_KHADAS_MCU is not set
 # CONFIG_MFD_QCOM_PM8008 is not set
 # CONFIG_MFD_INTEL_M10_BMC_SPI is not set
+CONFIG_MFD_RP1=y
 # CONFIG_MFD_LS2K_BMC_CORE is not set
 # CONFIG_MFD_RSMU_I2C is not set
 # CONFIG_MFD_RSMU_SPI is not set
@@ -5102,6 +5134,7 @@ CONFIG_REGULATOR_QCOM_SPMI=y
 # CONFIG_REGULATOR_RAA215300 is not set
 # CONFIG_REGULATOR_RASPBERRYPI_TOUCHSCREEN_ATTINY is not set
 # CONFIG_REGULATOR_RASPBERRYPI_TOUCHSCREEN_V2 is not set
+# CONFIG_REGULATOR_WAVESHARE_TOUCHSCREEN is not set
 CONFIG_REGULATOR_RK808=y
 # CONFIG_REGULATOR_RT4801 is not set
 # CONFIG_REGULATOR_RT4803 is not set
@@ -5228,6 +5261,7 @@ CONFIG_V4L_MEM2MEM_DRIVERS=y
 # Amphion drivers
 #
 # CONFIG_VIDEO_AMPHION_VPU is not set
+# CONFIG_VIDEO_BCM2835_UNICAM_LEGACY is not set
 
 #
 # Aspeed media platform drivers
@@ -5294,6 +5328,7 @@ CONFIG_V4L_MEM2MEM_DRIVERS=y
 #
 # CONFIG_VIDEO_QCOM_CAMSS is not set
 # CONFIG_VIDEO_QCOM_IRIS is not set
+# CONFIG_VIDEO_RPI_HEVC_DEC is not set
 
 #
 # Raspberry Pi media platform drivers
@@ -5372,11 +5407,13 @@ CONFIG_VIDEOBUF2_VMALLOC=m
 # Camera ISPs
 #
 # CONFIG_VIDEO_THP7312 is not set
+# CONFIG_VIDEO_AD5398 is not set
 # end of Camera ISPs
 
 CONFIG_VIDEO_CAMERA_LENS=y
 # CONFIG_VIDEO_AD5820 is not set
 # CONFIG_VIDEO_AK7375 is not set
+# CONFIG_VIDEO_BU64754 is not set
 # CONFIG_VIDEO_DW9714 is not set
 # CONFIG_VIDEO_DW9719 is not set
 # CONFIG_VIDEO_DW9768 is not set
@@ -5445,6 +5482,7 @@ CONFIG_VIDEO_CAMERA_LENS=y
 # CONFIG_VIDEO_TW9903 is not set
 # CONFIG_VIDEO_TW9906 is not set
 # CONFIG_VIDEO_TW9910 is not set
+# CONFIG_VIDEO_IRS1125 is not set
 # CONFIG_VIDEO_VPX3220 is not set
 
 #
@@ -5552,7 +5590,7 @@ CONFIG_DRM_CLIENT_DEFAULT_FBDEV=y
 CONFIG_DRM_CLIENT_DEFAULT="fbdev"
 # end of Supported DRM clients
 
-# CONFIG_DRM_LOAD_EDID_FIRMWARE is not set
+CONFIG_DRM_LOAD_EDID_FIRMWARE=y
 CONFIG_DRM_DISPLAY_DP_AUX_BUS=y
 CONFIG_DRM_DISPLAY_HELPER=y
 CONFIG_DRM_BRIDGE_CONNECTOR=y
@@ -5610,6 +5648,8 @@ CONFIG_DRM_AMDGPU=m
 # Display Engine Configuration
 #
 CONFIG_DRM_AMD_DC=y
+CONFIG_DRM_AMD_DC_FP=y
+# CONFIG_DRM_AMD_SECURE_DISPLAY is not set
 # end of Display Engine Configuration
 
 # CONFIG_HSA_AMD is not set
@@ -5687,6 +5727,7 @@ CONFIG_DRM_PANEL=y
 # CONFIG_DRM_PANEL_ILITEK_ILI9341 is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9805 is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9806E is not set
+# CONFIG_DRM_PANEL_ILITEK_ILI9806E_SPI is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9881C is not set
 # CONFIG_DRM_PANEL_ILITEK_ILI9882T is not set
 # CONFIG_DRM_PANEL_INNOLUX_EJ030NA is not set
@@ -5765,6 +5806,7 @@ CONFIG_DRM_PANEL_SIMPLE=y
 # CONFIG_DRM_PANEL_SUMMIT is not set
 # CONFIG_DRM_PANEL_SYNAPTICS_R63353 is not set
 # CONFIG_DRM_PANEL_TDO_TL070WSH30 is not set
+# CONFIG_DRM_PANEL_TPO_Y17P is not set
 # CONFIG_DRM_PANEL_TPO_TD028TTEC1 is not set
 # CONFIG_DRM_PANEL_TPO_TD043MTEA1 is not set
 # CONFIG_DRM_PANEL_TPO_TPG110 is not set
@@ -5774,6 +5816,8 @@ CONFIG_DRM_PANEL_SIMPLE=y
 # CONFIG_DRM_PANEL_VISIONOX_RM69299 is not set
 # CONFIG_DRM_PANEL_VISIONOX_RM692E5 is not set
 # CONFIG_DRM_PANEL_VISIONOX_VTDR6130 is not set
+# CONFIG_DRM_PANEL_WAVESHARE_TOUCHSCREEN is not set
+# CONFIG_DRM_PANEL_WAVESHARE_TOUCHSCREEN_V2 is not set
 # CONFIG_DRM_PANEL_WIDECHIPS_WS2401 is not set
 # CONFIG_DRM_PANEL_XINPENG_XPP055C272 is not set
 # end of Display Panels
@@ -5848,9 +5892,9 @@ CONFIG_DRM_DW_MIPI_DSI=y
 # CONFIG_DRM_IMX8_DC is not set
 # CONFIG_DRM_IMX_DCSS is not set
 # CONFIG_DRM_IMX_LCDC is not set
-CONFIG_DRM_V3D=m
-CONFIG_DRM_VC4=m
-# CONFIG_DRM_VC4_HDMI_CEC is not set
+CONFIG_DRM_V3D=y
+CONFIG_DRM_VC4=y
+CONFIG_DRM_VC4_HDMI_CEC=y
 # CONFIG_DRM_ETNAVIV is not set
 CONFIG_DRM_HISI_HIBMC=m
 CONFIG_DRM_HISI_KIRIN=y
@@ -5879,7 +5923,7 @@ CONFIG_DRM_MESON_DW_MIPI_DSI=y
 # CONFIG_DRM_XEN_FRONTEND is not set
 # CONFIG_DRM_LIMA is not set
 CONFIG_DRM_PANFROST=m
-CONFIG_DRM_PANTHOR=m
+# CONFIG_DRM_PANTHOR is not set
 # CONFIG_DRM_TIDSS is not set
 # CONFIG_DRM_GUD is not set
 # CONFIG_DRM_ST7571_I2C is not set
@@ -5896,6 +5940,10 @@ CONFIG_DRM_PANEL_ORIENTATION_QUIRKS=y
 # Frame buffer Devices
 #
 # CONFIG_FB is not set
+# CONFIG_FB_BCM2708 is not set
+# CONFIG_FB_IMX is not set
+# CONFIG_FB_EFI is not set
+CONFIG_FB_RPISENSE=y
 CONFIG_FB_CORE=y
 # CONFIG_FB_DEVICE is not set
 CONFIG_FB_SYS_FILLRECT=y
@@ -5918,6 +5966,7 @@ CONFIG_BACKLIGHT_CLASS_DEVICE=y
 # CONFIG_BACKLIGHT_KTD2801 is not set
 # CONFIG_BACKLIGHT_KTZ8866 is not set
 # CONFIG_BACKLIGHT_PWM is not set
+# CONFIG_BACKLIGHT_RPI is not set
 # CONFIG_BACKLIGHT_QCOM_WLED is not set
 # CONFIG_BACKLIGHT_ADP8860 is not set
 # CONFIG_BACKLIGHT_ADP8870 is not set
@@ -5955,7 +6004,7 @@ CONFIG_LOGO_LINUX_CLUT224=y
 # CONFIG_TRACE_GPU_MEM is not set
 # end of Graphics support
 
-CONFIG_DRM_ACCEL=y
+# CONFIG_DRM_ACCEL is not set
 CONFIG_DRM_ACCEL_ROCKET=m
 CONFIG_SOUND=m
 CONFIG_SND=m
@@ -6006,6 +6055,7 @@ CONFIG_SND_USB_AUDIO_USE_MEDIA_CONTROLLER=y
 # CONFIG_SND_USB_VARIAX is not set
 CONFIG_SND_SOC=m
 CONFIG_SND_SOC_GENERIC_DMAENGINE_PCM=y
+# CONFIG_SND_SOC_ADI is not set
 # CONFIG_SND_SOC_USB is not set
 
 #
@@ -6020,6 +6070,7 @@ CONFIG_SND_SOC_GENERIC_DMAENGINE_PCM=y
 #
 # CONFIG_SND_SOC_AMD_ACP is not set
 # CONFIG_SND_AMD_ACP_CONFIG is not set
+# CONFIG_SND_ATMEL_SOC is not set
 # end of AMD
 
 #
@@ -6043,6 +6094,48 @@ CONFIG_SND_SOC_GENERIC_DMAENGINE_PCM=y
 #
 # CONFIG_SND_BCM2835_SOC_I2S is not set
 # CONFIG_SND_BCM63XX_I2S_WHISTLER is not set
+# CONFIG_SND_BCM2708_SOC_CHIPDIP_DAC is not set
+# CONFIG_SND_BCM2708_SOC_GOOGLEVOICEHAT_SOUNDCARD is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_ADC is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_ADC8X is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DAC is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUS is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSHD is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSADC is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSADCPRO is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DACPLUSDSP is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_DIGI is not set
+# CONFIG_SND_BCM2708_SOC_HIFIBERRY_AMP is not set
+# CONFIG_SND_BCM2708_SOC_PIFI_40 is not set
+# CONFIG_SND_BCM2708_SOC_RPI_CIRRUS is not set
+# CONFIG_SND_BCM2708_SOC_RPI_DAC is not set
+# CONFIG_SND_BCM2708_SOC_RPI_PROTO is not set
+# CONFIG_SND_BCM2708_SOC_JUSTBOOM_BOTH is not set
+# CONFIG_SND_BCM2708_SOC_JUSTBOOM_DAC is not set
+# CONFIG_SND_BCM2708_SOC_JUSTBOOM_DIGI is not set
+# CONFIG_SND_BCM2708_SOC_IQAUDIO_CODEC is not set
+# CONFIG_SND_BCM2708_SOC_IQAUDIO_DAC is not set
+# CONFIG_SND_BCM2708_SOC_IQAUDIO_DIGI is not set
+# CONFIG_SND_BCM2708_SOC_I_SABRE_Q2M is not set
+# CONFIG_SND_BCM2708_SOC_ADAU1977_ADC is not set
+# CONFIG_SND_AUDIOINJECTOR_PI_SOUNDCARD is not set
+# CONFIG_SND_AUDIOINJECTOR_OCTO_SOUNDCARD is not set
+# CONFIG_SND_AUDIOINJECTOR_ISOLATED_SOUNDCARD is not set
+# CONFIG_SND_AUDIOSENSE_PI is not set
+# CONFIG_SND_DIGIDAC1_SOUNDCARD is not set
+# CONFIG_SND_BCM2708_SOC_DIONAUDIO_LOCO is not set
+# CONFIG_SND_BCM2708_SOC_DIONAUDIO_LOCO_V2 is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_PIANO_DAC is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_PIANO_DAC_PLUS is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_BOSS_DAC is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_BOSS2_DAC is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_DIGIONE is not set
+# CONFIG_SND_BCM2708_SOC_ALLO_KATANA_DAC is not set
+# CONFIG_SND_BCM2708_SOC_FE_PI_AUDIO is not set
+# CONFIG_SND_PISOUND is not set
+# CONFIG_SND_RPI_SIMPLE_SOUNDCARD is not set
+# CONFIG_SND_RPI_WM8804_SOUNDCARD is not set
+# CONFIG_SND_DACBERRY400 is not set
 # end of Broadcom
 
 #
@@ -6150,6 +6243,8 @@ CONFIG_SND_SOC_GENERIC_DMAENGINE_PCM=y
 # end of PXA
 
 # CONFIG_SND_SOC_QCOM is not set
+# CONFIG_SND_RP1_AUDIO_OUT is not set
+# CONFIG_SND_SOC_ROCKCHIP is not set
 
 #
 # Renesas
@@ -6234,6 +6329,8 @@ CONFIG_SND_SOC_I2C_AND_SPI=m
 # CODEC drivers
 #
 # CONFIG_SND_SOC_AC97_CODEC is not set
+# CONFIG_SND_SOC_AD193X_SPI is not set
+# CONFIG_SND_SOC_AD193X_I2C is not set
 # CONFIG_SND_SOC_ADAU1372_I2C is not set
 # CONFIG_SND_SOC_ADAU1372_SPI is not set
 # CONFIG_SND_SOC_ADAU1373 is not set
@@ -6315,6 +6412,7 @@ CONFIG_SND_SOC_HDMI_CODEC=m
 # CONFIG_SND_SOC_ICS43432 is not set
 # CONFIG_SND_SOC_IDT821034 is not set
 # CONFIG_SND_SOC_INNO_RK3036 is not set
+# CONFIG_SND_SOC_MA120X0P is not set
 # CONFIG_SND_SOC_MAX98088 is not set
 # CONFIG_SND_SOC_MAX98090 is not set
 # CONFIG_SND_SOC_MAX98357A is not set
@@ -6447,6 +6545,7 @@ CONFIG_SND_SOC_HDMI_CODEC=m
 # CONFIG_SND_SOC_LPASS_VA_MACRO is not set
 # CONFIG_SND_SOC_LPASS_RX_MACRO is not set
 # CONFIG_SND_SOC_LPASS_TX_MACRO is not set
+# CONFIG_SND_SOC_I_SABRE_CODEC is not set
 # end of CODEC drivers
 
 #
@@ -6464,25 +6563,25 @@ CONFIG_HID_SUPPORT=y
 CONFIG_HID=y
 # CONFIG_HID_BATTERY_STRENGTH is not set
 CONFIG_HIDRAW=y
-CONFIG_UHID=m
+# CONFIG_UHID is not set
 CONFIG_HID_GENERIC=y
 # CONFIG_HID_HAPTIC is not set
 
 #
 # Special HID drivers
 #
-CONFIG_HID_A4TECH=m
+CONFIG_HID_A4TECH=y
 # CONFIG_HID_ACCUTOUCH is not set
 # CONFIG_HID_ACRUX is not set
-CONFIG_HID_APPLE=m
+CONFIG_HID_APPLE=y
 # CONFIG_HID_APPLEIR is not set
 # CONFIG_HID_ASUS is not set
 # CONFIG_HID_AUREAL is not set
-CONFIG_HID_BELKIN=m
+CONFIG_HID_BELKIN=y
 # CONFIG_HID_BETOP_FF is not set
 # CONFIG_HID_BIGBEN_FF is not set
-CONFIG_HID_CHERRY=m
-CONFIG_HID_CHICONY=m
+CONFIG_HID_CHERRY=y
+CONFIG_HID_CHICONY=y
 # CONFIG_HID_CORSAIR is not set
 # CONFIG_HID_COUGAR is not set
 # CONFIG_HID_MACALLY is not set
@@ -6490,14 +6589,14 @@ CONFIG_HID_CHICONY=m
 # CONFIG_HID_CMEDIA is not set
 # CONFIG_HID_CP2112 is not set
 # CONFIG_HID_CREATIVE_SB0540 is not set
-CONFIG_HID_CYPRESS=m
+CONFIG_HID_CYPRESS=y
 # CONFIG_HID_DRAGONRISE is not set
 # CONFIG_HID_EMS_FF is not set
 # CONFIG_HID_ELAN is not set
 # CONFIG_HID_ELECOM is not set
 # CONFIG_HID_ELO is not set
 # CONFIG_HID_EVISION is not set
-CONFIG_HID_EZKEY=m
+CONFIG_HID_EZKEY=y
 # CONFIG_HID_FT260 is not set
 # CONFIG_HID_GEMBIRD is not set
 # CONFIG_HID_GFRM is not set
@@ -6515,17 +6614,17 @@ CONFIG_HID_EZKEY=m
 # CONFIG_HID_VIEWSONIC is not set
 # CONFIG_HID_VRC2 is not set
 # CONFIG_HID_XIAOMI is not set
-CONFIG_HID_GYRATION=m
+CONFIG_HID_GYRATION=y
 # CONFIG_HID_ICADE is not set
-CONFIG_HID_ITE=m
+CONFIG_HID_ITE=y
 # CONFIG_HID_JABRA is not set
 # CONFIG_HID_TWINHAN is not set
-CONFIG_HID_KENSINGTON=m
+CONFIG_HID_KENSINGTON=y
 # CONFIG_HID_LCPOWER is not set
 # CONFIG_HID_LED is not set
 # CONFIG_HID_LENOVO is not set
 # CONFIG_HID_LETSKETCH is not set
-CONFIG_HID_LOGITECH=m
+CONFIG_HID_LOGITECH=y
 # CONFIG_HID_LOGITECH_DJ is not set
 # CONFIG_HID_LOGITECH_HIDPP is not set
 CONFIG_LOGITECH_FF=y
@@ -6537,17 +6636,17 @@ CONFIG_LOGIWHEELS_FF=y
 # CONFIG_HID_MAYFLASH is not set
 # CONFIG_HID_MEGAWORLD_FF is not set
 CONFIG_HID_REDRAGON=m
-CONFIG_HID_MICROSOFT=m
-CONFIG_HID_MONTEREY=m
+CONFIG_HID_MICROSOFT=y
+CONFIG_HID_MONTEREY=y
 CONFIG_HID_MULTITOUCH=m
 # CONFIG_HID_NINTENDO is not set
 # CONFIG_HID_NTI is not set
 # CONFIG_HID_NTRIG is not set
 # CONFIG_HID_ORTEK is not set
-CONFIG_HID_PANTHERLORD=m
+CONFIG_HID_PANTHERLORD=y
 CONFIG_PANTHERLORD_FF=y
 # CONFIG_HID_PENMOUNT is not set
-CONFIG_HID_PETALYNX=m
+CONFIG_HID_PETALYNX=y
 # CONFIG_HID_PICOLCD is not set
 # CONFIG_HID_PLANTRONICS is not set
 # CONFIG_HID_PLAYSTATION is not set
@@ -6557,20 +6656,20 @@ CONFIG_HID_PETALYNX=m
 # CONFIG_HID_RETRODE is not set
 # CONFIG_HID_ROCCAT is not set
 # CONFIG_HID_SAITEK is not set
-CONFIG_HID_SAMSUNG=m
+CONFIG_HID_SAMSUNG=y
 # CONFIG_HID_SEMITEK is not set
 # CONFIG_HID_SIGMAMICRO is not set
 # CONFIG_HID_SONY is not set
 # CONFIG_HID_SPEEDLINK is not set
 # CONFIG_HID_STEAM is not set
 # CONFIG_HID_STEELSERIES is not set
-CONFIG_HID_SUNPLUS=m
+CONFIG_HID_SUNPLUS=y
 # CONFIG_HID_RMI is not set
 # CONFIG_HID_GREENASIA is not set
 # CONFIG_HID_HYPERV_MOUSE is not set
 # CONFIG_HID_SMARTJOYPLUS is not set
 # CONFIG_HID_TIVO is not set
-CONFIG_HID_TOPSEED=m
+CONFIG_HID_TOPSEED=y
 # CONFIG_HID_TOPRE is not set
 # CONFIG_HID_THINGM is not set
 # CONFIG_HID_THRUSTMASTER is not set
@@ -6603,7 +6702,7 @@ CONFIG_I2C_HID=m
 #
 # USB HID support
 #
-CONFIG_USB_HID=m
+CONFIG_USB_HID=y
 # CONFIG_HID_PID is not set
 CONFIG_USB_HIDDEV=y
 
@@ -6646,11 +6745,11 @@ CONFIG_USB_DEFAULT_AUTHORIZATION_MODE=1
 # USB Host Controller Drivers
 #
 # CONFIG_USB_C67X00_HCD is not set
-CONFIG_USB_XHCI_HCD=m
+CONFIG_USB_XHCI_HCD=y
 # CONFIG_USB_XHCI_DBGCAP is not set
-CONFIG_USB_XHCI_PCI=m
+CONFIG_USB_XHCI_PCI=y
 # CONFIG_USB_XHCI_PCI_RENESAS is not set
-CONFIG_USB_XHCI_PLATFORM=m
+CONFIG_USB_XHCI_PLATFORM=y
 # CONFIG_USB_XHCI_HISTB is not set
 # CONFIG_USB_XHCI_MTK is not set
 # CONFIG_USB_XHCI_MVEBU is not set
@@ -6676,6 +6775,7 @@ CONFIG_USB_OHCI_HCD_PLATFORM=y
 # CONFIG_USB_SL811_HCD is not set
 # CONFIG_USB_R8A66597_HCD is not set
 # CONFIG_USB_RENESAS_USBHS_HCD is not set
+# CONFIG_USB_DWCOTG is not set
 # CONFIG_USB_HCD_TEST_MODE is not set
 # CONFIG_USB_XEN_HCD is not set
 CONFIG_USB_RENESAS_USBHS=y
@@ -6691,7 +6791,7 @@ CONFIG_USB_WDM=m
 #
 # NOTE: USB_STORAGE depends on SCSI but BLK_DEV_SD may also be needed; see USB_STORAGE Help for more info
 #
-CONFIG_USB_STORAGE=m
+CONFIG_USB_STORAGE=y
 # CONFIG_USB_STORAGE_DEBUG is not set
 CONFIG_USB_STORAGE_REALTEK=m
 CONFIG_REALTEK_AUTOPM=y
@@ -6707,7 +6807,7 @@ CONFIG_USB_STORAGE_ONETOUCH=m
 CONFIG_USB_STORAGE_KARMA=m
 CONFIG_USB_STORAGE_CYPRESS_ATACB=m
 CONFIG_USB_STORAGE_ENE_UB6250=m
-CONFIG_USB_UAS=m
+CONFIG_USB_UAS=y
 
 #
 # USB Imaging devices
@@ -6785,13 +6885,13 @@ CONFIG_USB_SERIAL_GENERIC=y
 # CONFIG_USB_SERIAL_AIRCABLE is not set
 # CONFIG_USB_SERIAL_ARK3116 is not set
 # CONFIG_USB_SERIAL_BELKIN is not set
-CONFIG_USB_SERIAL_CH341=m
+CONFIG_USB_SERIAL_CH341=y
 # CONFIG_USB_SERIAL_WHITEHEAT is not set
 # CONFIG_USB_SERIAL_DIGI_ACCELEPORT is not set
-CONFIG_USB_SERIAL_CP210X=m
+CONFIG_USB_SERIAL_CP210X=y
 # CONFIG_USB_SERIAL_CYPRESS_M8 is not set
 # CONFIG_USB_SERIAL_EMPEG is not set
-CONFIG_USB_SERIAL_FTDI_SIO=m
+CONFIG_USB_SERIAL_FTDI_SIO=y
 # CONFIG_USB_SERIAL_VISOR is not set
 # CONFIG_USB_SERIAL_IPAQ is not set
 # CONFIG_USB_SERIAL_IR is not set
@@ -6812,7 +6912,7 @@ CONFIG_USB_SERIAL_FTDI_SIO=m
 # CONFIG_USB_SERIAL_MOS7840 is not set
 # CONFIG_USB_SERIAL_MXUPORT is not set
 # CONFIG_USB_SERIAL_NAVMAN is not set
-CONFIG_USB_SERIAL_PL2303=m
+CONFIG_USB_SERIAL_PL2303=y
 # CONFIG_USB_SERIAL_OTI6858 is not set
 # CONFIG_USB_SERIAL_QCAUX is not set
 # CONFIG_USB_SERIAL_QUALCOMM is not set
@@ -7011,6 +7111,8 @@ CONFIG_MMC_BLOCK=y
 CONFIG_MMC_BLOCK_MINORS=32
 # CONFIG_SDIO_UART is not set
 # CONFIG_MMC_TEST is not set
+# CONFIG_MMC_BCM2835_MMC is not set
+# CONFIG_MMC_BCM2835_SDHOST is not set
 
 #
 # MMC/SD/SDIO Host Controller Drivers
@@ -7019,27 +7121,27 @@ CONFIG_MMC_BLOCK_MINORS=32
 CONFIG_MMC_ARMMMCI=y
 CONFIG_MMC_QCOM_DML=y
 CONFIG_MMC_STM32_SDMMC=y
-CONFIG_MMC_SDHCI=m
+CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_IO_ACCESSORS=y
 CONFIG_MMC_SDHCI_UHS2=m
-CONFIG_MMC_SDHCI_PCI=m
+CONFIG_MMC_SDHCI_PCI=y
 CONFIG_MMC_RICOH_MMC=y
-CONFIG_MMC_SDHCI_ACPI=m
-CONFIG_MMC_SDHCI_PLTFM=m
-CONFIG_MMC_SDHCI_OF_ARASAN=m
+CONFIG_MMC_SDHCI_ACPI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ARASAN=y
 # CONFIG_MMC_SDHCI_OF_AT91 is not set
-CONFIG_MMC_SDHCI_OF_ESDHC=m
-CONFIG_MMC_SDHCI_OF_DWCMSHC=m
-CONFIG_MMC_SDHCI_CADENCE=m
+CONFIG_MMC_SDHCI_OF_ESDHC=y
+CONFIG_MMC_SDHCI_OF_DWCMSHC=y
+CONFIG_MMC_SDHCI_CADENCE=y
 # CONFIG_MMC_SDHCI_ESDHC_IMX is not set
-CONFIG_MMC_SDHCI_TEGRA=m
+CONFIG_MMC_SDHCI_TEGRA=y
 # CONFIG_MMC_SDHCI_PXAV3 is not set
-CONFIG_MMC_SDHCI_F_SDH30=m
+CONFIG_MMC_SDHCI_F_SDH30=y
 # CONFIG_MMC_SDHCI_MILBEAUT is not set
-CONFIG_MMC_SDHCI_IPROC=m
+CONFIG_MMC_SDHCI_IPROC=y
 CONFIG_MMC_MESON_GX=y
 # CONFIG_MMC_MESON_MX_SDIO is not set
-CONFIG_MMC_SDHCI_MSM=m
+CONFIG_MMC_SDHCI_MSM=y
 # CONFIG_MMC_MXC is not set
 # CONFIG_MMC_TIFM_SD is not set
 CONFIG_MMC_SPI=y
@@ -7069,8 +7171,8 @@ CONFIG_MMC_CQHCI=y
 # CONFIG_MMC_TOSHIBA_PCI is not set
 CONFIG_MMC_BCM2835=y
 # CONFIG_MMC_MTK is not set
-CONFIG_MMC_SDHCI_BRCMSTB=m
-CONFIG_MMC_SDHCI_XENON=m
+CONFIG_MMC_SDHCI_BRCMSTB=y
+CONFIG_MMC_SDHCI_XENON=y
 # CONFIG_MMC_SDHCI_AM654 is not set
 # CONFIG_MMC_OWL is not set
 CONFIG_SCSI_UFSHCD=m
@@ -7169,11 +7271,13 @@ CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
 #
 # CONFIG_LEDS_TRIGGER_TRANSIENT is not set
 # CONFIG_LEDS_TRIGGER_CAMERA is not set
+# CONFIG_LEDS_TRIGGER_INPUT is not set
 # CONFIG_LEDS_TRIGGER_PANIC is not set
 # CONFIG_LEDS_TRIGGER_NETDEV is not set
 # CONFIG_LEDS_TRIGGER_PATTERN is not set
 # CONFIG_LEDS_TRIGGER_TTY is not set
 # CONFIG_LEDS_TRIGGER_INPUT_EVENTS is not set
+# CONFIG_LEDS_TRIGGER_ACTPWR is not set
 
 #
 # Simatic LED drivers
@@ -7250,6 +7354,7 @@ CONFIG_RTC_INTF_DEV=y
 # CONFIG_RTC_DRV_ABB5ZES3 is not set
 # CONFIG_RTC_DRV_ABEOZ9 is not set
 # CONFIG_RTC_DRV_ABX80X is not set
+CONFIG_RTC_DRV_RPI=y
 CONFIG_RTC_DRV_BRCMSTB=y
 CONFIG_RTC_DRV_DS1307=y
 # CONFIG_RTC_DRV_DS1307_CENTURY is not set
@@ -7402,6 +7507,7 @@ CONFIG_PL330_DMA=y
 CONFIG_TEGRA186_GPC_DMA=y
 CONFIG_TEGRA20_APB_DMA=y
 CONFIG_TEGRA210_ADMA=y
+CONFIG_DMA_BCM2708=y
 # CONFIG_XGENE_DMA is not set
 # CONFIG_XILINX_DMA is not set
 # CONFIG_XILINX_XDMA is not set
@@ -7494,17 +7600,17 @@ CONFIG_VMGENID=m
 # CONFIG_ARM_CCA_GUEST is not set
 CONFIG_VIRTIO_ANCHOR=y
 CONFIG_VIRTIO=y
-CONFIG_VIRTIO_PCI_LIB=m
-CONFIG_VIRTIO_PCI_LIB_LEGACY=m
+CONFIG_VIRTIO_PCI_LIB=y
+CONFIG_VIRTIO_PCI_LIB_LEGACY=y
 CONFIG_VIRTIO_MENU=y
-CONFIG_VIRTIO_PCI=m
+CONFIG_VIRTIO_PCI=y
 CONFIG_VIRTIO_PCI_LEGACY=y
 CONFIG_VIRTIO_VDPA=m
 CONFIG_VIRTIO_PMEM=y
-CONFIG_VIRTIO_BALLOON=m
+CONFIG_VIRTIO_BALLOON=y
 # CONFIG_VIRTIO_MEM is not set
-CONFIG_VIRTIO_INPUT=m
-CONFIG_VIRTIO_MMIO=m
+CONFIG_VIRTIO_INPUT=y
+CONFIG_VIRTIO_MMIO=y
 CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_VIRTIO_DMA_SHARED_BUFFER=m
 # CONFIG_VIRTIO_DEBUG is not set
@@ -7602,6 +7708,7 @@ CONFIG_COMMON_CLK=y
 # CONFIG_COMMON_CLK_MAX9485 is not set
 CONFIG_COMMON_CLK_RK808=y
 CONFIG_COMMON_CLK_RP1=y
+CONFIG_COMMON_CLK_RP1_SDIO=y
 CONFIG_COMMON_CLK_HI655X=y
 CONFIG_COMMON_CLK_SCMI=y
 CONFIG_COMMON_CLK_SCPI=y
@@ -7819,6 +7926,7 @@ CONFIG_COMMON_CLK_MESON_VID_PLL_DIV=y
 CONFIG_COMMON_CLK_MESON_VCLK=y
 CONFIG_COMMON_CLK_MESON_CLKC_UTILS=y
 CONFIG_COMMON_CLK_MESON_AO_CLKC=y
+CONFIG_COMMON_CLK_MESON_EE_CLKC=y
 CONFIG_COMMON_CLK_MESON_CPU_DYNDIV=y
 CONFIG_COMMON_CLK_GXBB=y
 CONFIG_COMMON_CLK_AXG=y
@@ -8126,6 +8234,8 @@ CONFIG_TEGRA_IOMMU_SMMU=y
 # CONFIG_MTK_IOMMU is not set
 # CONFIG_VIRTIO_IOMMU is not set
 # CONFIG_SPRD_IOMMU is not set
+CONFIG_MBOX_RP1=y
+CONFIG_BCM2712_IOMMU=y
 
 #
 # Remoteproc drivers
@@ -9032,9 +9142,10 @@ CONFIG_PWM=y
 # CONFIG_PWM_ATMEL_TCB is not set
 # CONFIG_PWM_AXI_PWMGEN is not set
 CONFIG_PWM_BCM2835=y
+CONFIG_PWM_PIO_RP1=y
 CONFIG_PWM_BCM_IPROC=y
 # CONFIG_PWM_BERLIN is not set
-# CONFIG_PWM_BRCMSTB is not set
+CONFIG_PWM_BRCMSTB=y
 # CONFIG_PWM_CLK is not set
 # CONFIG_PWM_DWC is not set
 # CONFIG_PWM_FSL_FTM is not set
@@ -9050,6 +9161,8 @@ CONFIG_PWM_MESON=y
 # CONFIG_PWM_OMAP_DMTIMER is not set
 # CONFIG_PWM_PCA9685 is not set
 CONFIG_PWM_RASPBERRYPI_POE=y
+CONFIG_PWM_RP1=y
+# CONFIG_PWM_RCAR is not set
 # CONFIG_PWM_RENESAS_RCAR is not set
 # CONFIG_PWM_RENESAS_TPU is not set
 CONFIG_PWM_ROCKCHIP=y
@@ -9258,7 +9371,7 @@ CONFIG_PHY_TEGRA194_P2U=y
 # CONFIG_PHY_XILINX_ZYNQMP is not set
 # end of PHY Subsystem
 
-CONFIG_POWERCAP=y
+# CONFIG_POWERCAP is not set
 # CONFIG_IDLE_INJECT is not set
 # CONFIG_ARM_SCMI_POWERCAP is not set
 # CONFIG_DTPM is not set
@@ -9288,6 +9401,7 @@ CONFIG_ARM64_BRBE=y
 # CONFIG_ARM_DMC620_PMU is not set
 # CONFIG_MARVELL_CN10K_TAD_PMU is not set
 # CONFIG_ALIBABA_UNCORE_DRW_PMU is not set
+# CONFIG_RPI_AXIPERF is not set
 CONFIG_HISI_PMU=m
 CONFIG_HISI_PCIE_PMU=m
 CONFIG_HNS3_PMU=m
@@ -9331,6 +9445,7 @@ CONFIG_NVMEM_LAYOUTS=y
 # CONFIG_NVMEM_LAYOUT_SL28_VPD is not set
 # CONFIG_NVMEM_LAYOUT_ONIE_TLV is not set
 # CONFIG_NVMEM_LAYOUT_U_BOOT_ENV is not set
+CONFIG_NVMEM_RASPBERRYPI_OTP=y
 # end of Layout Types
 
 CONFIG_NVMEM_BCM_OCOTP=y
@@ -9390,6 +9505,7 @@ CONFIG_FS_STACK=y
 CONFIG_BUFFER_HEAD=y
 CONFIG_LEGACY_DIRECT_IO=y
 # CONFIG_EXT2_FS is not set
+# CONFIG_EXT3_FS is not set
 CONFIG_EXT4_FS=y
 CONFIG_EXT4_USE_FOR_EXT2=y
 CONFIG_EXT4_FS_POSIX_ACL=y
@@ -9422,8 +9538,10 @@ CONFIG_BTRFS_FS_POSIX_ACL=y
 # CONFIG_BTRFS_DEBUG is not set
 # CONFIG_BTRFS_ASSERT is not set
 # CONFIG_BTRFS_EXPERIMENTAL is not set
+# CONFIG_BTRFS_FS_REF_VERIFY is not set
 # CONFIG_NILFS2_FS is not set
 # CONFIG_F2FS_FS is not set
+# CONFIG_BCACHEFS_FS is not set
 # CONFIG_FS_DAX is not set
 CONFIG_FS_POSIX_ACL=y
 CONFIG_EXPORTFS=y
@@ -9753,8 +9871,30 @@ CONFIG_INTEGRITY_TRUSTED_KEYRING=y
 CONFIG_INTEGRITY_PLATFORM_KEYRING=y
 CONFIG_LOAD_UEFI_KEYS=y
 CONFIG_INTEGRITY_AUDIT=y
-# CONFIG_IMA is not set
-# CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT is not set
+CONFIG_IMA=y
+# CONFIG_IMA_KEXEC is not set
+CONFIG_IMA_MEASURE_PCR_IDX=10
+CONFIG_IMA_LSM_RULES=y
+CONFIG_IMA_NG_TEMPLATE=y
+# CONFIG_IMA_SIG_TEMPLATE is not set
+CONFIG_IMA_DEFAULT_TEMPLATE="ima-ng"
+# CONFIG_IMA_DEFAULT_HASH_SHA1 is not set
+# CONFIG_IMA_DEFAULT_HASH_SHA256 is not set
+CONFIG_IMA_DEFAULT_HASH_SHA512=y
+CONFIG_IMA_DEFAULT_HASH="sha512"
+CONFIG_IMA_WRITE_POLICY=y
+CONFIG_IMA_READ_POLICY=y
+CONFIG_IMA_APPRAISE=y
+CONFIG_IMA_ARCH_POLICY=y
+# CONFIG_IMA_APPRAISE_BUILD_POLICY is not set
+CONFIG_IMA_APPRAISE_BOOTPARAM=y
+# CONFIG_IMA_APPRAISE_MODSIG is not set
+# CONFIG_IMA_BLACKLIST_KEYRING is not set
+# CONFIG_IMA_LOAD_X509 is not set
+CONFIG_IMA_MEASURE_ASYMMETRIC_KEYS=y
+CONFIG_IMA_QUEUE_EARLY_BOOT_KEYS=y
+CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT=y
+# CONFIG_IMA_DISABLE_HTABLE is not set
 # CONFIG_EVM is not set
 # CONFIG_DEFAULT_SECURITY_SELINUX is not set
 # CONFIG_DEFAULT_SECURITY_APPARMOR is not set
@@ -9776,6 +9916,8 @@ CONFIG_CC_HAS_AUTO_VAR_INIT_ZERO=y
 CONFIG_INIT_STACK_ALL_ZERO=y
 CONFIG_CC_HAS_SANCOV_STACK_DEPTH_CALLBACK=y
 CONFIG_KSTACK_ERASE=y
+CONFIG_GCC_PLUGIN_STACKLEAK=y
+# CONFIG_GCC_PLUGIN_STACKLEAK_VERBOSE is not set
 CONFIG_KSTACK_ERASE_TRACK_MIN_SIZE=100
 # CONFIG_KSTACK_ERASE_METRICS is not set
 # CONFIG_KSTACK_ERASE_RUNTIME_DISABLE is not set
@@ -9803,6 +9945,7 @@ CONFIG_BUG_ON_DATA_CORRUPTION=y
 CONFIG_CC_HAS_RANDSTRUCT=y
 CONFIG_RANDSTRUCT_NONE=y
 # CONFIG_RANDSTRUCT_FULL is not set
+# CONFIG_RANDSTRUCT_PERFORMANCE is not set
 # end of Kernel hardening options
 # end of Security options
 
@@ -9861,6 +10004,7 @@ CONFIG_CRYPTO_ECC=y
 CONFIG_CRYPTO_ECDH=y
 # CONFIG_CRYPTO_ECDSA is not set
 # CONFIG_CRYPTO_ECRDSA is not set
+# CONFIG_CRYPTO_CURVE25519 is not set
 # end of Public-key cryptography
 
 #
@@ -10118,6 +10262,9 @@ CONFIG_CRYPTO_LIB_UTILS=y
 CONFIG_CRYPTO_LIB_AES=y
 CONFIG_CRYPTO_LIB_ARC4=y
 CONFIG_CRYPTO_LIB_GF128MUL=y
+CONFIG_CRYPTO_LIB_BLAKE2S_GENERIC=y
+CONFIG_CRYPTO_ARCH_HAVE_LIB_CHACHA=y
+CONFIG_CRYPTO_LIB_CHACHA_GENERIC=y
 CONFIG_CRYPTO_LIB_CHACHA=y
 CONFIG_CRYPTO_LIB_CHACHA_ARCH=y
 CONFIG_CRYPTO_LIB_CURVE25519=y
@@ -10128,6 +10275,7 @@ CONFIG_CRYPTO_LIB_POLY1305=y
 CONFIG_CRYPTO_LIB_POLY1305_ARCH=y
 CONFIG_CRYPTO_LIB_POLY1305_GENERIC=y
 CONFIG_CRYPTO_LIB_POLY1305_RSIZE=9
+CONFIG_CRYPTO_ARCH_HAVE_LIB_POLY1305=y
 CONFIG_CRYPTO_LIB_CHACHA20POLY1305=y
 CONFIG_CRYPTO_LIB_SHA1=y
 CONFIG_CRYPTO_LIB_SHA1_ARCH=y
@@ -10136,6 +10284,8 @@ CONFIG_CRYPTO_LIB_SHA256_ARCH=y
 CONFIG_CRYPTO_LIB_SHA512=y
 CONFIG_CRYPTO_LIB_SHA512_ARCH=y
 CONFIG_CRYPTO_LIB_SM3=y
+CONFIG_CRYPTO_CHACHA20_NEON=y
+CONFIG_CRYPTO_POLY1305_NEON=y
 # end of Crypto library routines
 
 CONFIG_XXHASH=y
@@ -10235,7 +10385,9 @@ CONFIG_OID_REGISTRY=y
 CONFIG_UCS2_STRING=y
 CONFIG_HAVE_GENERIC_VDSO=y
 CONFIG_GENERIC_GETTIMEOFDAY=y
+CONFIG_GENERIC_VDSO_TIME_NS=y
 CONFIG_VDSO_GETRANDOM=y
+CONFIG_GENERIC_VDSO_DATA_STORE=y
 CONFIG_FONT_SUPPORT=y
 CONFIG_FONTS=y
 # CONFIG_FONT_8x8 is not set
@@ -10257,6 +10409,7 @@ CONFIG_MEMREGION=y
 CONFIG_ARCH_HAS_UACCESS_FLUSHCACHE=y
 CONFIG_ARCH_STACKWALK=y
 CONFIG_STACKDEPOT=y
+CONFIG_STACKDEPOT_ALWAYS_INIT=y
 CONFIG_STACKDEPOT_MAX_FRAMES=64
 CONFIG_SBITMAP=y
 CONFIG_PARMAN=m
@@ -10315,7 +10468,9 @@ CONFIG_DEBUG_INFO_BTF_MODULES=y
 # CONFIG_GDB_SCRIPTS is not set
 CONFIG_FRAME_WARN=2048
 # CONFIG_STRIP_ASM_SYMS is not set
+# CONFIG_READABLE_ASM is not set
 # CONFIG_HEADERS_INSTALL is not set
+# CONFIG_DEBUG_SECTION_MISMATCH is not set
 CONFIG_SECTION_MISMATCH_WARN_ONLY=y
 # CONFIG_DEBUG_FORCE_FUNCTION_ALIGN_64B is not set
 CONFIG_ARCH_WANT_FRAME_POINTERS=y
@@ -10340,11 +10495,15 @@ CONFIG_HAVE_ARCH_KGDB=y
 CONFIG_ARCH_HAS_UBSAN=y
 CONFIG_UBSAN=y
 # CONFIG_UBSAN_TRAP is not set
+CONFIG_CC_HAS_UBSAN_BOUNDS_STRICT=y
 CONFIG_CC_HAS_UBSAN_ARRAY_BOUNDS=y
 CONFIG_UBSAN_BOUNDS=y
+CONFIG_UBSAN_BOUNDS_STRICT=y
 CONFIG_UBSAN_ARRAY_BOUNDS=y
 CONFIG_UBSAN_SHIFT=y
+# CONFIG_UBSAN_DIV_ZERO is not set
 # CONFIG_UBSAN_UNREACHABLE is not set
+CONFIG_UBSAN_SIGNED_WRAP=y
 CONFIG_UBSAN_BOOL=y
 CONFIG_UBSAN_ENUM=y
 # CONFIG_UBSAN_ALIGNMENT is not set
@@ -10371,6 +10530,7 @@ CONFIG_PAGE_EXTENSION=y
 # CONFIG_DEBUG_PAGEALLOC is not set
 CONFIG_SLUB_DEBUG=y
 # CONFIG_SLUB_DEBUG_ON is not set
+# CONFIG_SLUB_RCU_DEBUG is not set
 # CONFIG_PAGE_OWNER is not set
 CONFIG_PAGE_TABLE_CHECK=y
 CONFIG_PAGE_TABLE_CHECK_ENFORCED=y
@@ -10405,7 +10565,12 @@ CONFIG_HAVE_ARCH_KASAN_VMALLOC=y
 CONFIG_CC_HAS_KASAN_GENERIC=y
 CONFIG_CC_HAS_KASAN_SW_TAGS=y
 CONFIG_CC_HAS_WORKING_NOSANITIZE_ADDRESS=y
-# CONFIG_KASAN is not set
+CONFIG_KASAN=y
+# CONFIG_KASAN_GENERIC is not set
+# CONFIG_KASAN_SW_TAGS is not set
+CONFIG_KASAN_HW_TAGS=y
+CONFIG_KASAN_VMALLOC=y
+# CONFIG_KASAN_EXTRA_INFO is not set
 CONFIG_HAVE_ARCH_KFENCE=y
 CONFIG_KFENCE=y
 CONFIG_KFENCE_SAMPLE_INTERVAL=100
@@ -10421,6 +10586,7 @@ CONFIG_KFENCE_STRESS_TEST_FAULTS=0
 # Debug Oops, Lockups and Hangs
 #
 CONFIG_PANIC_ON_OOPS=y
+CONFIG_PANIC_ON_OOPS_VALUE=1
 CONFIG_PANIC_TIMEOUT=-1
 # CONFIG_SOFTLOCKUP_DETECTOR is not set
 CONFIG_HAVE_HARDLOCKUP_DETECTOR_BUDDY=y
@@ -10460,6 +10626,7 @@ CONFIG_LOCK_DEBUGGING_SUPPORT=y
 
 # CONFIG_DEBUG_IRQFLAGS is not set
 CONFIG_STACKTRACE=y
+# CONFIG_WARN_ALL_UNSEEDED_RANDOM is not set
 # CONFIG_DEBUG_KOBJECT is not set
 
 #
diff --git a/kernel/prepare/pkg.yaml b/kernel/prepare/pkg.yaml
index 6c6e61d..27c599b 100644
--- a/kernel/prepare/pkg.yaml
+++ b/kernel/prepare/pkg.yaml
@@ -6,8 +6,8 @@ dependencies:
   - image: "{{ .LLVM_IMAGE }}:{{ .TOOLS_REV }}"
 steps:
   - sources:
-      - url: https://cdn.kernel.org/pub/linux/kernel/v{{ regexReplaceAll "(\\d+)(.\\d+)(\\.\\d+)?$" .linux_version "${1}" }}.x/linux-{{ .linux_version }}.tar.xz
-        destination: linux.tar.xz
+      - url: "https://github.com/raspberrypi/linux/archive/{{ .linux_version }}.tar.gz"
+        destination: linux.tar.gz
         sha256: "{{ .linux_sha256 }}"
         sha512: "{{ .linux_sha512 }}"
       - url: https://github.com/a13xp0p0v/kernel-hardening-checker/archive/{{ .kspp_ref }}.tar.gz
@@ -19,7 +19,7 @@ steps:
     prepare:
       - |
         mkdir -p /src
-        tar -xJf linux.tar.xz --strip-components=1 -C /src
+        tar -xzf linux.tar.gz --strip-components=1 -C /src
 
         cd /src
         make mrproper
